_test/issue-1189.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+type I interface {
+	Foo() int
+}
+
+type S1 struct {
+	i int
+}
+
+func (s S1) Foo() int { return s.i }
+
+type S2 struct{}
+
+func (s *S2) Foo() int { return 42 }
+
+func main() {
+	Is := map[string]I{
+		"foo": S1{21},
+		"bar": &S2{},
+	}
+	n := 0
+	for _, s := range Is {
+		n += s.Foo()
+	}
+	println(n, Is["foo"].Foo())
```

Comment:
With this suggestion we also test that the part that was removed L1670 is ok, and not just the part that was removed L1625.
```suggestion
	bar := "bar"
	println(n, Is["foo"].Foo(), Is[bar].Foo())
```

Reply:
applied